lib/charliecloud.py

```diff
@@ -1191,6 +1371,74 @@ def lstrip(self, n):
       return Path(".").joinpath(*self.parts[n:])
 
+   def mkdir_(self):
+      TRACE("ensuring directory: %s" % self)
+      try:
+         super().mkdir(exist_ok=True)
+      except FileExistsError as x:
+          FATAL("can't mkdir: exists and not a directory: %s" % x.filename)
+      except OSError as x:
+         FATAL("can't mkdir: %s: %s: %s" % (self.name, x.filename, x.strerror))
+
+   def mkdirs(self, exist_ok=True):
+      TRACE("ensuring directories: %s" % self.name)
+      try:
+         os.makedirs(self, exist_ok=exist_ok)
+      except OSError as x:
+         FATAL("can't mkdir: %s: %s: %s" % (self.name, x.filename, x.strerror))
+         
+   def open(self, mode, *args, **kwargs):
+      "Error-checking wrapper for open()."
+      return ossafe(super().open, "can't open for %s: %s" % (mode, self.name),
+                    mode, *args, **kwargs) # note: removed 'self' from front of this line
+
+   def rename(self, name_new):
+      if (Path(name_new).exists()): # is this check necessary? necessary to use ossafe here?
+         FATAL("can't rename: destination exists: %s" % name_new)
+      ossafe(super().rename, "can't rename: %s -> %s" % (self.name, name_new),
+             name_new)
+
+   def rmdir(self):
+      ossafe(super().rmdir, "can't rmdir: %s" % self.name)
+
+   def rmtree(self):
+      if (self.is_dir()):
+         TRACE("deleting directory: %s" % self.name)
+         try:
+            shutil.rmtree(self)
+         except OSError as x:
+            FATAL("can't recursively delete directory %s: %s: %s"
+                  % (self.name, x.filename, x.strerror))
+      else:
+         assert False, "unimplemented"
+
+   def stat_(self, links=False):
+      """An error-checking version of stat(). Note that we cannot simply change
+         the definition of stat() to be ossafe, as the exists() method in pathlib
+         relies on an OSError check.    
+         See: https://github.com/python/cpython/blob/3.10/Lib/pathlib.py#L1291
+         NOTE: We also cannot just call super().stat here because the follow_symlinks
+         kwarg is absent in pathlib for Python 3.6, which we want to retain 
+         compatibility with."""
+      return ossafe(os.stat, "can't stat: %s" % self.name, self, follow_symlinks=links)
+
+   def symlink(self, target, clobber=False):
+      if (clobber and self.is_file()):
+         self.unlink_()
+      try:
+         super().symlink_to(target)
+      except FileExistsError:
+         if (not self.is_symlink()):
+            FATAL("can't symlink: source exists and isn't a symlink: %s" % self.name)
+         if (self.readlink() != target):
+            FATAL("can't symlink: %s exists; want target %s but existing is %s"
+                  % (self.name, target, self.readlink()))
+      except OSError as x:
+         FATAL("can't symlink: %s -> %s: %s" % (self.name, target, x.strerror))
+
+   def unlink_(self, *args, **kwargs):
+      "Error-checking wrapper for unlink method"
```

Comment:
Can probably omit docstrings like this; it can be implied by the trailing underscore.